contrib/win32/openssh/ssh-add-hostkey.ps1

```diff
@@ -0,0 +1,73 @@
+<# 
+ Author: manoj.ampalam@microsoft.com
+
+ Description: ssh-add.exe like Powershell utility to do host key management.
+ Input parameter mimic ssh-add.exe cmdline arguments. 
+ 
+ Host keys on Windows need to be registered as SYSTEM (i.e ssh-add.exe would 
+ need to run as SYSTEM while talking to ssh-agent). This typically requires 
+ an external utility like psexec. 
+
+ This script tries to use the Task scheduler option:
+  - registers a task scheduler task to run ssh-add.exe operation as SYSTEM
+  - actual output of ssh-add.exe is written to file (both stdout and stderr)
+  - Dumps the file contents to console
+
+#>
+
+# see https://linux.die.net/man/1/ssh-add for what the arguments mean
+Param(
+  [switch]$List_fingerprints, #ssh-add -l 
+  [switch]$List_pubkeys,      #ssh-add -L 
+  [switch]$Delete_key,        #ssh-add -d 
+  [switch]$Delete_all,        #ssh-add -D
+  [string]$key="" 
+
+)
+
+$switch_count = 0
```

Comment:
use [parameterset](https://msdn.microsoft.com/en-us/library/dd878348(v=vs.85).aspx) to exclude those parameters for different scenarios and it makes sure those switch does not show together when you hit [tab]